src/clr-angular/data/datagrid/providers/columns.service.ts

```diff
@@ -3,43 +3,32 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import { Injectable, OnDestroy } from '@angular/core';
-import { BehaviorSubject, Subscription } from 'rxjs';
-import { DatagridColumnState } from '../interfaces/column-state.interface';
-import { DatagridRenderStep } from '../enums/render-step.enum';
-import { DatagridRenderOrganizer } from '../render/render-organizer';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { ColumnStateDiff, DatagridColumnState } from '../interfaces/column-state.interface';
 
 @Injectable()
-export class ColumnsService implements OnDestroy {
-  subscriptions: Subscription[] = [];
+export class ColumnsService {
   columns: BehaviorSubject<DatagridColumnState>[] = [];
 
-  constructor(private organizer: DatagridRenderOrganizer) {
-    this.subscriptions.push(
-      this.organizer.filterRenderSteps(DatagridRenderStep.CLEAR_WIDTHS).subscribe(() => this.reset())
```

Comment:
I don't remember why this is being removed.

Reply:
This callback of `DatagridRenderStep.CLEAR_WIDTHS` get called with a change of `{width: 0}`. But instantly after that, the state is again overridden from the callback of `DatagridRenderStep.CLEAR_WIDTHS` in `HeaderRenderer`. So this was totally unnecessary. Even though the value gets overridden instantly, it was still calling the dom renderer.

Reply:
I recall talking to Eudes about this, and this was duplicating the behavior we had before. There was a reason related to ensuring the calculations are correct. @hippee-lee can you verify if this can be removed from the original refactor?

Reply:
I believe that duplication was there because cells and columns used to be handled independently. IIR cells always removed the width and strict width css class that overrides the flex behavior because the rendering cycle would determine that from the column. I think it had to be this way for the use case when all columns had been resized except for the last one and we had to remove the fixed-width class so it would flex. 

Now that cells and columns are managed from a single source of truth for width state it looks like this can be removed. I manually tested with the dev app and it behaved as I expected it to.

Reply:
Thanks resolving.